### DISCUSSION TITLE
Instructions for trying out experimental Rust support

### DISCUSSION THREAD

**Discussion by jpangia** [github/codeql#17781] [Q&A]
I see CodeQL recently added experimental support for Rust, but it looks like it isn't included in the CodeQL cli yet.
Looking at the readme in https://github.com/github/codeql/tree/main/rust for build instructions, it says that if you don't have access to `semmle-code` repository, you should build Bazel manually. But then it says to run `bazel run @codeql//rust:rust-installer` in a `semmle-code` directory, which I do not have access to.

Is it possible to run the experimental Rust support without access to `semmle-code`? If so, could you provide more detailed instructions?

**Comment by rvermeulen** [comment]
Hi @jpangia,

It is great to see your enthusiasm for Rust support.
Rust support is still in early development so unfortunately you would have to wait a bit longer before you can experiment with it.

Stay tuned!

**Comment by coadaflorin** [comment]
Hey @jpangia, just wanted to let you know that we shipped support for Rust in CodeQL to [public preview](https://github.blog/changelog/2025-06-30-codeql-support-for-rust-now-in-public-preview/). Give it a go and [let us know](https://github.com/orgs/community/discussions/161754#discussioncomment-13627726) how things went for you!
